<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1249</width>
    <height>807</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <layout class="QGridLayout" name="gridLayout">
    <item row="3" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>1</number>
          </property>
          <item>
           <widget class="QTextEdit" name="code_textbox">
            <property name="html">
             <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;int main()&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;{&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;	&lt;/p&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;}&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
            </property>
            <property name="acceptRichText">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item alignment="Qt::AlignBottom">
           <widget class="QTextEdit" name="terminal_output">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="styleSheet">
             <string notr="true">font: 13pt &quot;Ubuntu&quot;;
background-color: rgb(255, 255, 255);
color: rgb(0, 0, 0);</string>
            </property>
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTabWidget" name="code_tabs">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="asm_tab">
           <attribute name="title">
            <string>Assembly</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QTextEdit" name="asm_textbox">
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="styleSheet">
               <string notr="true">background-color: rgb(158, 158, 158);</string>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="bin_tab">
           <attribute name="title">
            <string>Binary</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="0">
             <widget class="QTextEdit" name="bin_textbox">
              <property name="autoFillBackground">
               <bool>false</bool>
              </property>
              <property name="styleSheet">
               <string notr="true">background-color: rgb(158, 158, 158);</string>
              </property>
              <property name="readOnly">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="data_tabs">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="register_tab">
           <attribute name="title">
            <string>Registers</string>
           </attribute>
           <layout class="QFormLayout" name="formLayout">
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_2">
              <item row="1" column="0">
               <widget class="QLabel" name="label_2">
                <property name="text">
                 <string>x1 (ra)</string>
                </property>
               </widget>
              </item>
              <item row="14" column="0">
               <widget class="QLabel" name="label_15">
                <property name="text">
                 <string>x14 (a4)</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QLineEdit" name="x3reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="10" column="1">
               <widget class="QLineEdit" name="x10reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="8" column="0">
               <widget class="QLabel" name="label_9">
                <property name="text">
                 <string>x8 (s0/fp)</string>
                </property>
               </widget>
              </item>
              <item row="12" column="0">
               <widget class="QLabel" name="label_13">
                <property name="text">
                 <string>x12 (a2)</string>
                </property>
               </widget>
              </item>
              <item row="0" column="3">
               <widget class="QLineEdit" name="x16reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="7" column="0">
               <widget class="QLabel" name="label_8">
                <property name="text">
                 <string>x7 (t2)</string>
                </property>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="label">
                <property name="text">
                 <string>x0 (zero):</string>
                </property>
               </widget>
              </item>
              <item row="4" column="0">
               <widget class="QLabel" name="label_5">
                <property name="text">
                 <string>x4 (tp)</string>
                </property>
               </widget>
              </item>
              <item row="11" column="1">
               <widget class="QLineEdit" name="x11reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="6" column="0">
               <widget class="QLabel" name="label_7">
                <property name="text">
                 <string>x6 (t1)</string>
                </property>
               </widget>
              </item>
              <item row="5" column="0">
               <widget class="QLabel" name="label_6">
                <property name="text">
                 <string>x5 (t0)</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="label_4">
                <property name="text">
                 <string>x3 (gp)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QLineEdit" name="x1reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="7" column="1">
               <widget class="QLineEdit" name="x7reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="13" column="1">
               <widget class="QLineEdit" name="x13reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="11" column="0">
               <widget class="QLabel" name="label_12">
                <property name="text">
                 <string>x11 (a1)</string>
                </property>
               </widget>
              </item>
              <item row="6" column="1">
               <widget class="QLineEdit" name="x6reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="12" column="1">
               <widget class="QLineEdit" name="x12reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="9" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>x9 (s1)</string>
                </property>
               </widget>
              </item>
              <item row="10" column="0">
               <widget class="QLabel" name="label_11">
                <property name="text">
                 <string>x10 (a0)</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QLineEdit" name="x0reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="15" column="0">
               <widget class="QLabel" name="label_16">
                <property name="text">
                 <string>x15 (a5)</string>
                </property>
               </widget>
              </item>
              <item row="8" column="1">
               <widget class="QLineEdit" name="x8reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="label_3">
                <property name="text">
                 <string>x2 (sp)</string>
                </property>
               </widget>
              </item>
              <item row="4" column="1">
               <widget class="QLineEdit" name="x4reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="5" column="1">
               <widget class="QLineEdit" name="x5reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QLineEdit" name="x2reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="15" column="1">
               <widget class="QLineEdit" name="x15reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="14" column="1">
               <widget class="QLineEdit" name="x14reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="0" column="2">
               <widget class="QLabel" name="label_28">
                <property name="text">
                 <string>x16 (a6)</string>
                </property>
               </widget>
              </item>
              <item row="9" column="1">
               <widget class="QLineEdit" name="x9reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="13" column="0">
               <widget class="QLabel" name="label_14">
                <property name="text">
                 <string>x13 (a3)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="2">
               <widget class="QLabel" name="label_17">
                <property name="text">
                 <string>x17 (a7)</string>
                </property>
               </widget>
              </item>
              <item row="2" column="2">
               <widget class="QLabel" name="label_18">
                <property name="text">
                 <string>x18 (s2)</string>
                </property>
               </widget>
              </item>
              <item row="3" column="2">
               <widget class="QLabel" name="label_19">
                <property name="text">
                 <string>x19 (s3)</string>
                </property>
               </widget>
              </item>
              <item row="4" column="2">
               <widget class="QLabel" name="label_20">
                <property name="text">
                 <string>x20 (s4)</string>
                </property>
               </widget>
              </item>
              <item row="5" column="2">
               <widget class="QLabel" name="label_21">
                <property name="text">
                 <string>x21 (s5)</string>
                </property>
               </widget>
              </item>
              <item row="6" column="2">
               <widget class="QLabel" name="label_22">
                <property name="text">
                 <string>x22 (s6)</string>
                </property>
               </widget>
              </item>
              <item row="7" column="2">
               <widget class="QLabel" name="label_23">
                <property name="text">
                 <string>x23 (s7)</string>
                </property>
               </widget>
              </item>
              <item row="8" column="2">
               <widget class="QLabel" name="label_24">
                <property name="text">
                 <string>x24 (s8)</string>
                </property>
               </widget>
              </item>
              <item row="9" column="2">
               <widget class="QLabel" name="label_25">
                <property name="text">
                 <string>x25 (s9)</string>
                </property>
               </widget>
              </item>
              <item row="10" column="2">
               <widget class="QLabel" name="label_26">
                <property name="text">
                 <string>x26 (s10)</string>
                </property>
               </widget>
              </item>
              <item row="11" column="2">
               <widget class="QLabel" name="label_27">
                <property name="text">
                 <string>x27 (s11)</string>
                </property>
               </widget>
              </item>
              <item row="12" column="2">
               <widget class="QLabel" name="label_29">
                <property name="text">
                 <string>x28 (t3)</string>
                </property>
               </widget>
              </item>
              <item row="13" column="2">
               <widget class="QLabel" name="label_30">
                <property name="text">
                 <string>x29 (t4)</string>
                </property>
               </widget>
              </item>
              <item row="14" column="2">
               <widget class="QLabel" name="label_31">
                <property name="text">
                 <string>x30 (t5)</string>
                </property>
               </widget>
              </item>
              <item row="15" column="2">
               <widget class="QLabel" name="label_32">
                <property name="text">
                 <string>x31 (t6)</string>
                </property>
               </widget>
              </item>
              <item row="1" column="3">
               <widget class="QLineEdit" name="x17reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="2" column="3">
               <widget class="QLineEdit" name="x18reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="3" column="3">
               <widget class="QLineEdit" name="x19reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="4" column="3">
               <widget class="QLineEdit" name="x20reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="5" column="3">
               <widget class="QLineEdit" name="x21reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="6" column="3">
               <widget class="QLineEdit" name="x22reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="7" column="3">
               <widget class="QLineEdit" name="x23reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="8" column="3">
               <widget class="QLineEdit" name="x24reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="9" column="3">
               <widget class="QLineEdit" name="x25reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="10" column="3">
               <widget class="QLineEdit" name="x26reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="11" column="3">
               <widget class="QLineEdit" name="x27reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="12" column="3">
               <widget class="QLineEdit" name="x28reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="13" column="3">
               <widget class="QLineEdit" name="x29reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="14" column="3">
               <widget class="QLineEdit" name="x30reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
              <item row="15" column="3">
               <widget class="QLineEdit" name="x31reg">
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="memory_tab">
           <attribute name="title">
            <string>Memory</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_5">
              <item row="0" column="0">
               <widget class="QTableView" name="tableView">
                <property name="enabled">
                 <bool>false</bool>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="pushButton">
        <property name="text">
         <string>Compile</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="run_button">
        <property name="text">
         <string>Run</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
 <slots>
  <signal>signal1()</signal>
  <slot>slot1()</slot>
 </slots>
</ui>
